multi_partner_learning.py

```diff
@@ -70,36 +74,106 @@ def __init__(self,
         self.is_save_data = is_save_data
         self.save_folder = save_folder
         self.learning_computation_time = None
+        
 
         logger.debug("MultiPartnerLearning object instantiated.")
 
+    def train_single_model(self,partner):
+        #first check if the model was already trained
+        if not partner.model==None:
+            return partner.model
+        else:
+            # Initialize model
+            model = utils.generate_new_cnn_model()
+    
+            # Set if early stopping if needed
+            cb = []
+            if self.is_early_stopping:
+                es = EarlyStopping(monitor='val_loss', mode='min', verbose=0, patience=constants.PATIENCE)
+                cb.append(es)
+    
+            # Train model
+            logger.info(f"   Training model on partner with id #{partner.id}" )
+            model.fit(
+                partner.x_train,
+                partner.y_train,
+                batch_size=partner.batch_size,
+                epochs=self.epoch_count,
+                verbose=0,
+                validation_data=self.val_data,
+                callbacks=cb,
+            )
+            #store the model
+            partner.model=model
+            
+            self.nb_epochs_done = (es.stopped_epoch + 1) if es.stopped_epoch != 0 else self.epoch_count
```

Comment:
I assume the number of epochs can be different from a partner to another. Moreover, we maybe should differentiate the number of epochs of single training and the number used to train the meta_model for model stacking and keep all to analyze results of experiments.